src/components/terrain-panel.tsx

```diff
@@ -10,230 +10,277 @@ import { droughtLabels, DroughtLevel, terrainLabels, TerrainType, Vegetation, ve
-                  />
-                </div>
-                <div className={css.selector}>
-                  <DroughtSelector
-                    droughtLevel={zone.droughtLevel}
-                    onChange={this.handleDroughtChange}
-                    onChangeCommitted={this.handleDroughtChangeCommitted}
-                    severeDroughtAvailable={simulation.config.severeDroughtAvailable}
-                    disabled={simulation.config.droughtIndexLocked}
-                  />
-                </div>
-              </div>
-              <div className={css.buttonContainer}>
-                <Button className={css.continueButton} onClick={this.showNextPanel}>
-                  Next
-                </Button>
-              </div>
-            </div>
-            }
-            { currentPanel === 2 &&
-              <div className={css.panel}>
-                <div className={css.terrainTypeLabels}>{this.renderZoneTerrainTypeLabels()}</div>
-                <div className={css.terrainProperties}>{this.renderTerrainProperties()}</div>
-                <div className={css.wind}>
-                  <WindCircularControl />
-                </div>
-                <div className={css.buttonContainer}>
-                  <Button className={css.continueButton} onClick={this.showPreviousPanel}>
-                    Previous
-                  </Button>
-                  <Button className={css.continueButton} onClick={this.applyAndClose}>
-                    Create
-                  </Button>
-                </div>
-              </div>
-            }
-          </div>
-        }
-      </div>
-    );
-  }
+  }, [firstPanel, simulation.wind, simulation.zones, simulation.zones.length, simulation.zonesCount, ui.showTerrainUI]);
 
-  public handleClose = () => {
-    const { ui } = this.stores;
+  const setSelectedZone = (value: number) => {
+    ui.terrainUISelectedZone = value;
+  };
+
+  const handleClose = () => {
     ui.showTerrainUI = !ui.showTerrainUI;
     log("TerrainPanelClosed");
   };
 
-  public applyAndClose = () => {
-    const { ui, simulation } = this.stores;
+  const applyAndClose = () => {
     ui.showTerrainUI = !ui.showTerrainUI;
-    simulation.populateCellsData();
+    simulation.setWindSpeed(windSpeed);
+    simulation.setWindDirection(windDirection);
+    simulation.updateZones(zones);
     log("TerrainPanelSettingsSaved");
   };
 
-  public handleZoneChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+  const handleZoneChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     // Radio buttons always return string values. We're using hidden radio buttons to change selected zone
     const newZone = parseInt(event.target.value, 10);
-    if (newZone !== this.selectedZone) {
-      this.selectedZone = newZone;
+    if (newZone !== selectedZone) {
+      setSelectedZone(newZone);
     }
     log("TerrainPanelZoneChanged", { zone: newZone });
```

Comment:
Should this log message be within the if statement above it so it only logs when there is a change?